### PR TITLE
Refactor trycatch and classes from RzList to RzPVector

### PR DESCRIPTION
**Pull request by PeiweiHu**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

+ Refactor `trycatch` and `classes` from RzList to RzPVector.
+ Remove the deprecated API rz_bin_get_classes.

**Test plan**

CI is green.

**Closing issues**

partially address https://github.com/rizinorg/rizin/issues/3755
partially address https://github.com/rizinorg/rizin/issues/3775

### CODE REVIEW

**Review by XVilka — APPROVED** [review]

**Review by XVilka — CHANGES_REQUESTED** [review]
```
[XX] db/formats/pe/2331 PE: #2331
RZ_NOPLUGINS=1 /home/runner/bin/rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -eflirt.sigdb.load.system=false -eflirt.sigdb.load.home=false -N -A -Qc q! bins/pe/2331
-- stdout
--- expected
+++ actual
@@ -1,0 +1,1 @@
+WARNING: rz_bin_object_get_classes: assertion 'obj' failed (line 740)

-- stderr
ERROR: File is not PE
[ ] Analyze all flags starting with sym. and entry0 (aa)
[
[x] Analyze all flags starting with sym. and entry0 (aa)

[ ] Analyze function calls
[
[x] Analyze function calls
[ ] Analyze len bytes of instructions for references
[
[x] Analyze len bytes of instructions for references
[ ] Check for classes
[
[x] Check for classes
[ ] Analyze local variables and arguments
[
[x] Analyze local variables and arguments
[ ] Type matching analysis for all functions
[
[x] Type matching analysis for all functions
[ ] Applying signatures from sigdb
[
[x] Applied 0 FLIRT signatures via sigdb
[ ] Propagate noreturn information
[
[x] Propagate noreturn information
[ ] Integrate dwarf function information.
[
[x] Integrate dwarf function information.
[ ] Resolve pointers to data sections
[
[x] Resolve pointers to data sections

[x] Use -AA or aaaa to perform additional experimental analysis.

[XX] db/formats/pe/tinyXP PE: corkami tinyXP.exe - open
RZ_NOPLUGINS=1 /home/runner/bin/rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -eflirt.sigdb.load.system=false -eflirt.sigdb.load.home=false -N -A -Qc q! bins/pe/tinyXP.exe
-- stdout
--- expected
+++ actual
@@ -1,0 +1,1 @@
+WARNING: rz_bin_object_get_classes: assertion 'obj' failed (line 740)

-- stderr
ERROR: File is not PE
[ ] Analyze all flags starting with sym. and entry0 (aa)
[
[x] Analyze all flags starting with sym. and entry0 (aa)

[ ] Analyze function calls
[
[x] Analyze function calls
[ ] Analyze len bytes of instructions for references
[
[x] Analyze len bytes of instructions for references
[ ] Check for classes
[
[x] Check for classes
[ ] Analyze local variables and arguments
[
[x] Analyze local variables and arguments
[ ] Type matching analysis for all functions
[
[x] Type matching analysis for all functions
[ ] Applying signatures from sigdb
[
[x] Applied 0 FLIRT signatures via sigdb
[ ] Propagate noreturn information
[
[x] Propagate noreturn information
[ ] Integrate dwarf function information.
[
[x] Integrate dwarf function information.
[ ] Resolve pointers to data sections
[
[x] Resolve pointers to data sections

[x] Use -AA or aaaa to perform additional experimental analysis.

[XX] db/formats/dyldcache dyldcache CVE-[202](https://github.com/rizinorg/rizin/actions/runs/6192682267/job/16813115663?pr=3859#step:19:203)2-1[244](https://github.com/rizinorg/rizin/actions/runs/6192682267/job/16813115663?pr=3859#step:19:245)
RZ_NOPLUGINS=1 /home/runner/bin/rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -eflirt.sigdb.load.system=false -eflirt.sigdb.load.home=false -N -Qc 'aaa
q
' bins/dyldcache/CVE-2022-1244
-- stdout
--- expected
+++ actual
@@ -1,0 +1,1 @@
+WARNING: rz_bin_object_get_classes: assertion 'obj' failed (line 740)[XX] db/formats/pe/2331 PE: #2331
RZ_NOPLUGINS=1 /home/runner/bin/rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -eflirt.sigdb.load.system=false -eflirt.sigdb.load.home=false -N -A -Qc q! bins/pe/2331
-- stdout
--- expected
+++ actual
@@ -1,0 +1,1 @@
+WARNING: rz_bin_object_get_classes: assertion 'obj' failed (line 740)

-- stderr
ERROR: File is not PE
[ ] Analyze all flags starting with sym. and entry0 (aa)
[
[x] Analyze all flags starting with sym. and entry0 (aa)

[ ] Analyze function calls
[
[x] Analyze function calls
[ ] Analyze len bytes of instructions for references
[
[x] Analyze len bytes of instructions for references
[ ] Check for classes
[
[x] Check for classes
[ ] Analyze local variables and arguments
[
[x] Analyze local variables and arguments
[ ] Type matching analysis for all functions
[
[x] Type matching analysis for all functions
[ ] Applying signatures from sigdb
[
[x] Applied 0 FLIRT signatures via sigdb
[ ] Propagate noreturn information
[
[x] Propagate noreturn information
[ ] Integrate dwarf function information.
[
[x] Integrate dwarf function information.
[ ] Resolve pointers to data sections
[
[x] Resolve pointers to data sections

[x] Use -AA or aaaa to perform additional experimental analysis.

[XX] db/formats/dyldcache dyldcache CVE-[202](https://github.com/rizinorg/rizin/actions/runs/6192682267/job/16813115663?pr=3859#step:19:203)2-1[244](https://github.com/rizinorg/rizin/actions/runs/6192682267/job/16813115663?pr=3859#step:19:245)
RZ_NOPLUGINS=1 /home/runner/bin/rizin -escr.utf8=0 -escr.color=0 -escr.interactive=0 -eflirt.sigdb.load.system=false -eflirt.sigdb.load.home=false -N -Qc 'aaa
q
' bins/dyldcache/CVE-2022-1244
-- stdout
--- expected
+++ actual
@@ -1,0 +1,1 @@
+WARNING: rz_bin_object_get_classes: assertion 'obj' failed (line 740)
```

File: librz/bin/bfile.c [hunk 1]
```diff
@@ -499,7 +499,7 @@ RZ_API void rz_bin_class_free(RZ_NULLABLE RzBinClass *k) {
 	free(k);
 }
 
-RZ_API RzList /*<RzBinTrycatch *>*/ *rz_bin_file_get_trycatch(RZ_NONNULL RzBinFile *bf) {
+RZ_API RzPVector /*<RzBinTrycatch *>*/ *rz_bin_file_get_trycatch(RZ_NONNULL RzBinFile *bf) {
```
Comment (XVilka) [inline librz/bin/bfile.c]: Please add ownership

File: librz/bin/format/objc/mach0_classes.c [hunk 2]
```diff
@@ -1224,8 +1224,8 @@ static RzList *parse_swift_classes(RzBinFile *bf) {
 }
 #endif
 
-RZ_API RzList /*<RzBinClass *>*/ *MACH0_(parse_classes)(RzBinFile *bf, objc_cache_opt_info *oi) {
-	RzList /*<RzBinClass *>*/ *ret = NULL;
+RZ_API RzPVector /*<RzBinClass *>*/ *MACH0_(parse_classes)(RzBinFile *bf, objc_cache_opt_info *oi) {
```
Comment (XVilka) [inline librz/bin/format/objc/mach0_classes.c]: Ownership of the return value

File: librz/bin/format/objc/mach0_classes.h [hunk 3]
```diff
@@ -21,7 +21,7 @@
 #ifndef MACH0_CLASSES_H
 #define MACH0_CLASSES_H
 
-RZ_API RzList /*<RzBinClass *>*/ *MACH0_(parse_classes)(RzBinFile *bf, objc_cache_opt_info *oi);
+RZ_API RzPVector /*<RzBinClass *>*/ *MACH0_(parse_classes)(RzBinFile *bf, objc_cache_opt_info *oi);
```
Comment (XVilka) [inline librz/bin/format/objc/mach0_classes.h]: Ownership

File: librz/include/rz_bin.h [hunk 4]
```diff
@@ -923,12 +923,11 @@ RZ_DEPRECATE RZ_API RZ_BORROW RzList /*<RzBinAddr *>*/ *rz_bin_get_entries(RZ_NO
 RZ_DEPRECATE RZ_API RZ_BORROW RzList /*<RzBinField *>*/ *rz_bin_get_fields(RZ_NONNULL RzBin *bin);
 RZ_DEPRECATE RZ_API RZ_BORROW RzList /*<char *>*/ *rz_bin_get_libs(RZ_NONNULL RzBin *bin);
 RZ_DEPRECATE RZ_API RZ_BORROW RzList /*<RzBinSection *>*/ *rz_bin_get_sections(RZ_NONNULL RzBin *bin);
-RZ_DEPRECATE RZ_API RZ_BORROW RzList /*<RzBinClass *>*/ *rz_bin_get_classes(RZ_NONNULL RzBin *bin);
 RZ_DEPRECATE RZ_API RZ_BORROW RzList /*<RzBinString *>*/ *rz_bin_get_strings(RZ_NONNULL RzBin *bin);
 RZ_DEPRECATE RZ_API RZ_BORROW RzList /*<RzBinMem *>*/ *rz_bin_get_mem(RZ_NONNULL RzBin *bin);
 RZ_DEPRECATE RZ_API RZ_BORROW RzList /*<RzBinSymbol *>*/ *rz_bin_get_symbols(RZ_NONNULL RzBin *bin);
 RZ_DEPRECATE RZ_API int rz_bin_is_static(RZ_NONNULL RzBin *bin);
-RZ_API RzList /*<RzBinTrycatch *>*/ *rz_bin_file_get_trycatch(RZ_NONNULL RzBinFile *bf);
+RZ_API RzPVector /*<RzBinTrycatch *>*/ *rz_bin_file_get_trycatch(RZ_NONNULL RzBinFile *bf);
```
Comment (XVilka) [inline librz/include/rz_bin.h]: Ownership

**Review by XVilka — CHANGES_REQUESTED** [review]

**Comment by wargio** [comment]
nice changes! might speedup a lot stuff.

**Review by wargio — APPROVED** [review]

File: librz/analysis/class.c [hunk 5]
```diff
@@ -56,11 +56,17 @@ static const char *attr_type_id(RzAnalysisClassAttrType attr_type) {
 
 RZ_API void rz_analysis_class_recover_from_rzbin(RzAnalysis *analysis) {
 	rz_cons_break_push(NULL, NULL);
-	RzList *classes = rz_bin_get_classes(analysis->binb.bin);
+	RzBinObject *bin_obj = rz_bin_cur_object(analysis->binb.bin);
+	if (!bin_obj) {
+		rz_cons_break_pop();
```
Comment (wargio) [inline librz/analysis/class.c:61]: this shouldn't be here. only in core.

**Review by XVilka — APPROVED** [review]

**Pull request merged**